plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts

```diff
@@ -100,26 +102,43 @@ const createStepLogger = ({
   task: TaskContext;
   step: TaskStep;
 }) => {
-  const metadata = { stepId: step.id };
-  const taskLogger = winston.createLogger({
+  const stepLogStream = new PassThrough();
+  stepLogStream.on('data', async data => {
+    const message = data.toString().trim();
+    if (message?.length > 1) {
+      await task.emitLog(message, { stepId: step.id });
+    }
+  });
+
+  const taskLogger = WinstonLogger.create({
```

Comment:
Do we really need a winston logger here? I'm thinking it might be better to even avoid the extra fields provided by the `LoggerService` and just use a logger that just works with plain string messages. We're never using structured logging here anyway right? it's all gonna be flattened out and rendered in the frontend.

Got something like this in mind:

```ts
interface ScaffolderActionLogger {
  error(message: string, meta?: Error): void;
  warn(message: string): void;
  info(message: string): void;
  debug(message: string): void;
}
```

Reply:
Hmm, yeah, that's why I added some of the `TODO`s for us to do this later. I agree with you, but this would be a breaking change. Because it's the same logger that's used for client and backend logging, so not sure we can do this in a non-breaking way.